src/Block/NonceFieldBlock.php

```diff
@@ -36,7 +41,18 @@ public function __construct(
     ) {
         $this->_setNonce($nonce)
             ->_setFieldName($fieldName)
-            ->_setUseReferer($useReferer);
+            ->_setUseReferer($useReferer)
+            ->_setRefererFieldName(static::REFERER_FIELD_NAME);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    protected function _getRefererUrl()
+    {
+        return \esc_attr(\wp_unslash($_SERVER['REQUEST_URI']));
```

Comment:
I think that actually, this block is a specialization of a conceptual generic nonce block. So, yea, you can do that, but then add the WP layer.